flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/StateWithExecutionGraph.java

```diff
@@ -148,6 +162,33 @@ public Logger getLogger() {
         return logger;
     }
 
+    protected Throwable extractError(TaskExecutionStateTransition taskExecutionStateTransition) {
```

Comment:
We can get rid of this method once we unify `updateTaskExecutionState`

Reply:
After the change it would be used only within this class. I'd be OK with just making it private, so it's not exposed to the `AdaptiveScheduler`. But in general:

```
        final boolean successfulUpdate =
                getExecutionGraph().updateState(taskExecutionStateTransition);
        if (successfulUpdate
                && taskExecutionStateTransition.getExecutionState() == ExecutionState.FAILED) {
            final Throwable error = taskExecutionStateTransition.getError(classLoader);
            if (error == null) {
                error = new FlinkException("Unknown failure cause. Probably related to FLINK-21376.");
            }
            handleFailure(
                    Failure.createLocal(
                            error,
                            extractExecutionVertexID(taskExecutionStateTransition)));
        }
        return successfulUpdate;
```

is no less readable than

```
        final boolean successfulUpdate =
                getExecutionGraph().updateState(taskExecutionStateTransition);
        if (successfulUpdate
                && taskExecutionStateTransition.getExecutionState() == ExecutionState.FAILED) {
            handleFailure(
                    Failure.createLocal(
                            extractError(taskExecutionStateTransition),

                            extractExecutionVertexID(taskExecutionStateTransition)));
        }
        return successfulUpdate;
```

and it avoids additional noise / extra method (eg, when you read the code, you need to go to `extractError` to see what it really does). This will become especially visible when we get rid of the else clause.

But again, I have no strong feelings if we just change it to private.